### PR TITLE
fluentd review suggestions

### PR DESCRIPTION
**Pull request by thaJeztah**
suggestions from my review on https://github.com/moby/moby/pull/43100

- use a time.Duration for the configuration to make it less error-prone
- add a minimum and maximum range for the duration
- add a unit-test

@conorevans ptal; feel free to squash my commit with yours if these changes LGTY

### CODE REVIEW

File: daemon/logger/fluentd/fluentd.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ const (
 	defaultMaxRetries = math.MaxInt32
 	defaultRetryWait  = 1000
 
+	minReconnectInterval = 100 * time.Millisecond
+	maxReconnectInterval = 10 * time.Second
+
```
Comment (thaJeztah) [inline daemon/logger/fluentd/fluentd.go:53]: Wasn't sure if these were sensible min/max values, and/or if a "maximum" value makes sense (although I doubt an interval like 1 or 2 hours would be useful)

**Comment by conorevans** [comment]
Thanks @thaJeztah ! I'll merge this in and squash later (I haven't squashed -- manually, anyway -- in a while, so I'll read back up on that 😅 )

**Comment by thaJeztah** [comment]
Let me know if you need help on that! We used to have some docs about this, but they got moved to a different repo, and some of them are no longer there (but I can provide instructions if needed); old docs about rebasing and squashing can be found here; https://github.com/moby/moby/blob/v1.7.0/docs/project/work-issue.md#pull-and-rebase-frequently

Otherwise, I'm also happy to do the squashing for you (I think I should be able to push to your PR branch), so no worries if you don't know how 👍

**Comment by conorevans** [comment]
> Otherwise, I'm also happy to do the squashing for you (I think I should be able to push to your PR branch), so no worries if you don't know how 👍

Normally, I'd say "ah sure, good time to refresh / re-learn" but today I'm definitely happy for you to bail me out if you don't mind! Thanks 🙂

**Comment by thaJeztah** [comment]
> but today I'm definitely happy for you to bail me out if you don't mind! 

Sure! Feel free to merge this, and I'll do a squash after that.

Things may be somewhat slow during the Holidays (although maintainers may do some reviews in their spare time)

**Comment by conorevans** [comment]
Done @thaJeztah — ofc, totally get that :) there’s no rush.

**Pull request merged**